File: DSA/RATNA_KHANNA_590013141/DAY_55_QUESTION_2.c
#include <stdio.h>
#include <string.h>

int main() {
    char s[1000];
    printf("enter string\n");
    scanf("%s", s);
    int last[256];
    for (int i = 0; i < 256; i++) last[i] = -1;
    int start = 0, maxLen = 0;
    for (int i = 0; s[i]; i++) {
        if (last[(unsigned char)s[i]] >= start)
            start = last[(unsigned char)s[i]] + 1;
        last[(unsigned char)s[i]] = i;
        if (i - start + 1 > maxLen) maxLen = i - start + 1;
    }
    printf("output-%d", maxLen);
}

File: DSA/AnshulJain_590014934/Day_55_Question_2.py
s = input("Enter the string: ")
seen = {}
start = 0
max_len = 0
for i, ch in enumerate(s):
    if ch in seen and seen[ch] >= start:
        start = seen[ch] + 1
    seen[ch] = i
    max_len = max(max_len, i - start + 1)
print(max_len)
#for example:
# Input: s = "abcabcbb"
# Output: 3